src/mono/mono/utils/mono-codeman.c

```diff
@@ -689,5 +695,12 @@ mono_codeman_disable_write (void)
 		if (level == 0)
 			pthread_jit_write_protect_np (1);
 	}
+#elif defined(HOST_MACCAT) && defined(__aarch64__)
+		int level = GPOINTER_TO_INT (mono_native_tls_get_value (write_level_tls_id));
+		g_assert (level);
+		level --;
+		mono_native_tls_set_value (write_level_tls_id, GINT_TO_POINTER (level));
+		if (level == 0)
```

Comment:
I realize you're preserving the existing behavior, but having this counted seems weird.